bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/ConfigSecurityUtils.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.runtime.internal.configurable;
+
+import io.ballerina.runtime.api.creators.ErrorCreator;
+import io.ballerina.runtime.api.utils.StringUtils;
+import io.ballerina.runtime.internal.util.RuntimeUtils;
+import org.ballerinalang.config.cipher.AESCipherTool;
+import org.ballerinalang.config.cipher.AESCipherToolException;
+
+import java.io.IOException;
+import java.io.PrintStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static io.ballerina.runtime.internal.configurable.ConfigurableConstants.SECRET_FILE_NAME;
+
+/**
+ * Singleton class that holds security functions related to configurable values.
+ *
+ * @since 2.0.0
+ */
+public class ConfigSecurityUtils {
+
+    private static final Pattern encryptedFieldPattern = Pattern.compile("@encrypted:\\{(.*)\\}");
```

Comment:
patten needs to be `^@encrypted:\\{(.*)\\}$` otherwise the even strings with patten in the middle will match